src/sentry/api/bases/organization.py

```diff
@@ -146,29 +145,27 @@ def get_projects(
 
         user = getattr(request, 'user', None)
 
-        if (
-            user and is_active_superuser(request)
-            or include_allow_joinleave and organization.flags.allow_joinleave
-            or force_global_perms
-        ):
-            qs = Project.objects.filter(
-                organization=organization,
-                status=ProjectStatus.VISIBLE,
-            )
-        else:
-            qs = Project.objects.filter(
-                organization=organization,
-                teams__in=OrganizationMemberTeam.objects.filter(
-                    organizationmember__user=user,
-                    organizationmember__organization=organization,
-                ).values_list('team'),
-                status=ProjectStatus.VISIBLE,
-            )
+        qs = Project.objects.filter(
+            organization=organization,
+            status=ProjectStatus.VISIBLE,
+        )
 
         if project_ids:
             qs = qs.filter(id__in=project_ids)
 
-        projects = list(qs)
+        if force_global_perms:
+            projects = list(qs)
+        else:
+            if (
+                user and is_active_superuser(request) or
+                requested_projects or
+                include_allow_joinleave
+            ):
+                func = request.access.has_project_access
+            else:
```

Comment:
One thing you can do - If the request is membership you could bypass the queryset entirely and use access.projects for the core result set (then filter from there)

Reply:
yeah i kind of thought of that, but i was a little worried that it could be prone to future breakage since without doing some digging, it's not obvious what access.projects includes. what about doing something like adding additional methods on `access` that return projects based on membership / access? that could utilize the existing project set and seems like it'd be less confusing

Reply:
So you should always be able to trust access.projects include any project they are a member of based on the currently scoped request. For projects they have access to - realistically we don’t ever want to have to cache all of that since most of the time it’s every project in an org. That’s why I only made an access check vs also capturing the instances. 

We could probably still build helpers but it might make more sense on ProjectManager than within Access (though it could still hit access to double confirm the query didn’t return something bad). 

With all that said I’d still challenge why we’d ever need to return a list of every project a user potentially can access. I know there were product reasons but imo it’s a product deficiency if we require that (since it can’t scale due to it being unbounded). Similar concern with memberships but we know that list generally stays fairly small.